docs/reference/contributing/guidelines/workflow.md

```diff
@@ -50,53 +68,69 @@ Pull requests on GitHub have to meet the following requirements to keep the code
 - All new features and enhancements require documentation, tests and user guides for us to accept them. Please link each pull request to all relevant documentation and testing pull requests.
+- Avoid merging commmits. (Always rebase when possible.)
+- Pull requests should fix a bug or add a feature or refactor.
 
-#### Mbed OS maintainers
+#### Pull request categories
 
-The maintainers are a small group of Mbed OS engineers who are responsible for the Mbed OS codebase. Their primary role is to progress contributions, both internal and external, from the initial pull request state through to released code. They:
+##### Bug fixes
 
-1. Check for CLA compliance.
-2. Ensure the relevant stakeholders review pull requests.
-3. Guide contributors both technically and procedurally.
-4. Run pull requests through the CI systems.
-5. Merge pull requests into the requested branches.
-6. Make periodic patch and feature releases.
+The last line in your commit message description should say “Fixes #deadbeef”, where “deadbeef” is the issue number in GitHub. This allows GitHub to automatically close the issue when the commit merges into the default branch.
 
-The current maintainers are:
+Every bug fix should contain a test to verify results prior to and after the pull request.
+
+##### Changes and additions
+
+Backward compatible changes (such as refactoring and enhancements) or new target additions can go into patch and feature releases. We only consider features and new functionality additions for feature releases.
+
+##### Features
+
+We initially implement new features on separate branches in the Mbed OS repository. Mbed OS maintainers create the new branches by following the naming convention: "feature-" prefix.
+
+Each feature has a tech lead. This person is responsible for:
 
-* Anna Bridge (adbridge).
-* Martin Kojtal (0xc0170).
-* Jimmy Brisson (theotherjimmy).
-* Shrikant Tudavekar (studavekar).
-* Sam Grove (sg-).
+- Rebasing often to track master development.
+- Reviewing any addition to the feature branch (approval required by the feature tech lead or another assigned person).
 
 #### GitHub pull requests workflow
 
 Each pull request goes through the following workflow:
 
 <span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/Workflow.png)<span>The workflow of merging a pull request</span></span>
 
-##### Pull request states
+#### Pull request states
 
-Labels that the Mbed OS maintainers add to a pull request represent the pull request workflow states. The Mbed OS maintainers are responsible for moving pull requests through the workflow states.
+The Mbed OS maintainers add labels to a pull request that represent the pull request workflow states. The Mbed OS maintainers are responsible for moving pull requests through the workflow states.
+
+Each state is time boxed. In most cases, this is sufficient time to move to an another state.
```

Comment:
What happens when this time expires? this is not explained in the following sections.

Reply:
The actions will come as another update with more specification (once bot comes to play). For now, should be sufficient to explain that if an update is not provided within the time frame, pull request can be closed